components/modals/GoogleSignInModal.tsx

```diff
@@ -1,14 +1,99 @@
+import { useTheme } from '@react-navigation/native';
 import React from 'react';
-import { View } from 'react-native';
+import { Modal, Pressable, StyleSheet, TouchableOpacity, View } from 'react-native';
+import GoogleIcon from '../../assets/artworks/GoogleIcon';
+import { useAuth } from '../../context/auth';
+import Row from '../common/Row';
+import Space from '../common/Space';
+import StyledText from '../common/StyledText';
 
-// TODO: google sign-in modal
-/**
- * -  should show a Sign in with google button
- * -  clicking the button should close the modal and sign in the user
- */
+type GoogleSignInModalProps = {
+  visible: boolean;
+  onClose: () => void;
+};
+
+const GoogleSignInModal = ({ visible, onClose }: GoogleSignInModalProps) => {
+  const { signInWithGoogle } = useAuth();
+  const { colors } = useTheme();
+
+  const handleSignIn = () => {
+    signInWithGoogle();
+    onClose();
+  };
 
-const GoogleSignInModal = () => {
-  return <View />;
+  return (
+    <View style={styles.centeredView}>
+      <Modal animationType="slide" transparent visible={visible} onRequestClose={onClose}>
+        <View style={styles.centeredView}>
+          <View style={[styles.modalView, { backgroundColor: colors.bg }]}>
+            <Row style={styles.header}>
+              <StyledText> </StyledText>
+              <TouchableOpacity onPress={onClose}>
+                <StyledText style={{ color: colors.textLight }}>CANCEL</StyledText>
```

Comment:
- Could you use the same font as on design. 
- Also try to define the styles separately